From f3f138ce674be428ad1d723cad5b628d9f12816b Mon Sep 17 00:00:00 2001
From: Eric Erhardt <eric.erhardt@microsoft.com>
Date: Sat, 7 Oct 2017 22:04:43 -0500
Subject: [PATCH] Enable CLI to build on Windows during source-build. Don't
 build for net46 and use correct RID for core-setup.

---
 build/BundledRuntimes.props                                      | 2 +-
 src/Microsoft.DotNet.Cli.Utils/Microsoft.DotNet.Cli.Utils.csproj | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/build/BundledRuntimes.props b/build/BundledRuntimes.props
index 8a61a0c..260b8fd 100644
--- a/build/BundledRuntimes.props
+++ b/build/BundledRuntimes.props
@@ -1,7 +1,7 @@
 <Project>
   <PropertyGroup>
     <CoreSetupRid>$(HostRid)</CoreSetupRid>
-    <CoreSetupRid Condition=" '$(HostOSName)' == 'win' or '$(HostOSName)' == 'osx' ">$(HostMonikerRid)</CoreSetupRid>
+    <CoreSetupRid Condition=" ('$(HostOSName)' == 'win' or '$(HostOSName)' == 'osx') and '$(DotNetBuildFromSource)' != 'true' ">$(HostMonikerRid)</CoreSetupRid>
 
     <!-- only the runtime OSX .pkgs have a `-internal` suffix -->
     <InstallerStartSuffix Condition="'$(HostOSName)' == 'osx'">-internal</InstallerStartSuffix>
diff --git a/src/Microsoft.DotNet.Cli.Utils/Microsoft.DotNet.Cli.Utils.csproj b/src/Microsoft.DotNet.Cli.Utils/Microsoft.DotNet.Cli.Utils.csproj
index ed10567..de8e61f 100644
--- a/src/Microsoft.DotNet.Cli.Utils/Microsoft.DotNet.Cli.Utils.csproj
+++ b/src/Microsoft.DotNet.Cli.Utils/Microsoft.DotNet.Cli.Utils.csproj
@@ -4,7 +4,7 @@
   <PropertyGroup>
     <Version>$(SdkVersion)</Version>
     <TargetFrameworks>netstandard1.5;net46</TargetFrameworks>
-    <TargetFrameworks Condition=" '$(OS)' != 'Windows_NT' ">netstandard1.5</TargetFrameworks>
+    <TargetFrameworks Condition=" '$(OS)' != 'Windows_NT' or '$(DotNetBuildFromSource)' == 'true' ">netstandard1.5</TargetFrameworks>
     <WarningsAsErrors>true</WarningsAsErrors>
     <AssemblyOriginatorKeyFile>../../tools/Key.snk</AssemblyOriginatorKeyFile>
     <SignAssembly>true</SignAssembly>
-- 
1.9.1

